### PR TITLE
Ems 114809753 - Add video view data capture.

### PR DESCRIPTION
**Pull request by eschlange**
- Added new video view model.
- Added migration for new model.
- Added video javascript logic to capture the view.
- Added Vimeo event capturing lib file.
- Added a new route for video view capture.
- Added a controller to handle the view capture.
- Updated existing models with associations.
- Updated views to use the new video view capture logic.
- Moved inline video styles to css.
- Added specs for new model and controller.

[#114809753]

### CODE REVIEW

File: app/models/video.rb [hunk 1]
```diff
@@ -2,4 +2,6 @@
 class Video < ActiveRecord::Base
   has_and_belongs_to_many :users
   has_and_belongs_to_many :assignments
+
+  has_many :video_views, dependent: :destroy
```
Comment (ericcf) [inline app/models/video.rb]: The only issue here is that you probably don't want `VideoView` data to be destroyed even if a `Video` is destroyed. Any participant data should generally be preserved if possible. I'd probably go with `dependent: :restrict_with_exception` to make it harder to corrupt/destroy participant data.

File: config/routes.rb [hunk 2]
```diff
@@ -23,6 +23,9 @@
     get 'notify/:assignment_id/complete' =>
           'assignments#mark_assignment_complete_and_save',
         :as => 'notify_complete'
+    get 'notify/:video_id/viewed' =>
```
Comment (ericcf) [inline config/routes.rb]: Small change: I'd make this a `POST` because it results in creating a new record.

**Comment by ericcf** [comment]
With the two minor adjustments I suggested, I think it's ok to :ship:.

File: app/models/video.rb (hunk 1 above)
Comment (eschlange) [inline app/models/video.rb]: I agree with your sentiments. I'll update the parameter to use :restrict_with_exception.

File: config/routes.rb (hunk 2 above)
Comment (eschlange) [inline config/routes.rb]: Agreed, Rails makes it too easy to break REST conventions =D

**Comment by eschlange** [comment]
Updates complete, merging in changes.

**Pull request merged**